src/throttler/throttler.ts

```diff
@@ -145,52 +155,16 @@ export abstract class Throttler<T, R> {
     if (!taskData) {
       throw new Error(`taskData.get(${cursorIndex}) does not exist`);
     }
-    const task = taskData.task;
-    const tname = this.taskName(cursorIndex);
-    const t0 = Date.now();
+    const promises = [this.executeTask(taskData, cursorIndex)];
+    if (taskData.timeoutMillis) {
+      promises.push(this.initializeTimeout(taskData, cursorIndex));
+    }
 
     try {
-      const result = await this.handler(task);
-      const dt = Date.now() - t0;
-      if (dt < this.min) {
-        this.min = dt;
-      }
-      if (dt > this.max) {
-        this.max = dt;
-      }
-      this.avg = (this.avg * this.complete + dt) / (this.complete + 1);
-
-      this.success++;
-      this.complete++;
-      this.active--;
-      if (taskData.wait) {
-        taskData.wait.resolve(result);
-      }
-      this.taskDataMap.delete(cursorIndex);
-      this.process();
+      const result = await Promise.race(promises);
+      this.onTaskFulfilled(result, taskData, cursorIndex);
```

Comment:
If this fails throws for any reason, `task.onTaskFailed` will be run because this is still in a `try/catch` block. Is that intended?

Reply:
Good catch. I don't believe that's intended, so it should be moved out of the `try` block.

Reply:
Move `this.onTaskFulfilled()` out of the `try` block